File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
  vector<int>v;
  bool found =false;
  int binarySearch(vector<int>& nums,int start,int end,int target)
  {
    int i=start;
    int j=end;
    while(start<=end)
    {
        int mid=(start+end)/2;
        if(nums[mid]==target && i!=mid )
        {
           v.push_back(i+1);
           v.push_back(mid+1);
           found=true;
           return end;
        }
       else if(nums[mid]>target)
        {
            end=mid-1;
        }
        else
        {
            start=mid+1;
        }


    }
    return j;
    
  }


    vector<int> twoSum(vector<int>& numbers, int target) {
        int start=0;
        int end=numbers.size()-1;
        
        for(int i=0;i<numbers.size();i++)
        {
         
            end=binarySearch(numbers,i,end,target-numbers[i]);
            if(found)
            {
                break;
            }
            
        }
      return v;
    }
};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    int BinarySerach(int m ,int [] numbers ,int i)
    {
        int start=0;
        int end=numbers.length-1;
        int mid=(start+end)/2;
        while(start<=end)
        {
            mid=(start+end)/2;
            if( mid != i && numbers[mid]== m)
            {
                return mid;
            }
            else if(numbers[mid] >m)
            {
                end=mid-1;
            }
            else
            {
                start=mid+1;
            }
        }
        return -1;

    }
    public int[] twoSum(int[] numbers, int target) {
        int i=0;
        int [] arr=new int[2];
        while(i<numbers.length)
        {
            int m=target-numbers[i];
            int ind=BinarySerach(m ,numbers ,i);
             if( ind != -1 )
             {
               arr[0]=i+1;
               arr[1]=ind+1;
               break;
             }
             i++;
        }
        return arr;
    }
}